src/main/java/com/gradle/Utils.java

```diff
@@ -210,6 +210,26 @@ private static boolean isGradleNewerThan(String version) {
         return GradleVersion.current().compareTo(GradleVersion.version(version)) >= 0;
     }
 
+    /**
+     * Construct a repo URL from a git URL in the format of <code>git://github.com/acme-inc/my-project.git</code>. If
+     * the URL cannot be parsed, {@link Optional#empty()} is returned.
+     */
+    static Optional<String> extractRepoUrl(String gitUrl) {
```

Comment:
This is similar to the `toWebRepoUri` function in `CustomBuildScanEnhancements` can we re-use that function?

Reply:
Missed that, let me take a look

Reply:
Refactored it out to utils and kept the test if that looks alright to you. Also updated it to support `git://` schems. [e60c70f](https://github.com/gradle/common-custom-user-data-gradle-plugin/pull/169/commits/e60c70f4ca8b432640838c413a9049f57e6ab714)